vm/src/unix/os/cygwin_profil.c

```diff
@@ -0,0 +1,255 @@
+/* profil.c -- win32 profil.c equivalent
```

Comment:
It turns out that this function I borrowed actually is exported from a system library called `gmon` that isn't linked in by default. I'll explore using that instead of providing a replacement at some point.